src/server.rs

```diff
@@ -407,20 +410,57 @@ impl Server {
                         }),
                     );
 
+                    // Load OS root certificates, if enabled in the configuration.
+                    if config.general.trust_os_certificates {
+                        match get_os_root_certificates() {
+                            Ok(os_certs) => {
+                                let result = root_store.add_parsable_certificates(os_certs);
+
+                                debug!("{} get_os_root_certificates processed {} valid and {} invalid certs", address, result.0, result.1);
+                            }
+                            Err(err) => {
+                                warn!("{} Failed to load OS root certificates: {:?}", address, err)
+                            }
+                        }
+                    }
+
                     let mut tls_config = rustls::ClientConfig::builder()
                         .with_safe_defaults()
-                        .with_root_certificates(root_store)
+                        .with_root_certificates(root_store.clone())
                         .with_no_client_auth();
 
-                    // Equivalent to sslmode=prefer which is fine most places.
-                    // If you want verify-full, change `verify_server_certificate` to true.
-                    if !config.general.verify_server_certificate {
-                        let mut dangerous = tls_config.dangerous();
-                        dangerous.set_certificate_verifier(Arc::new(
-                            crate::tls::NoCertificateVerification {},
-                        ));
+                    {
+                        let mut dangerous = tls_config.dangerous(); // for security reasons, we will hide this variable from external scope
+
+                        match config.general.verify_server_certificate {
+                            CertificateVerificationVariant::Bool(v) => match v {
+                                true => { /* NOP */ } // verify-full (by default rustls checks certificates completely)
+                                false => {
+                                    // prefer
+                                    dangerous.set_certificate_verifier(Arc::new(
+                                        crate::tls::NoCertificateVerification {},
+                                    ));
+                                }
+                            },
+                            CertificateVerificationVariant::String(v) => match v.as_str() {
+                                "only-ca" => {
+                                    // verify-ca
+                                    dangerous.set_certificate_verifier(Arc::new(
+                                        crate::tls::OnlyRootCertificateVerification {
+                                            roots: root_store.clone(),
```

Comment:
This is a large structure, according to the docs. Can we pass a reference instead?

Reply:
> This is a large structure, according to the docs. Can we pass a reference instead?

I think yes, it can be converted to a reference